src/main/java/bio/terra/workspace/mdc/MdcLogEnhancerFilter.java

```diff
@@ -0,0 +1,35 @@
+package bio.terra.workspace.mdc;
+
+import java.io.IOException;
+import java.util.concurrent.ThreadLocalRandom;
+import javax.servlet.*;
+import javax.servlet.http.HttpServletRequest;
+import org.hashids.Hashids;
+import org.slf4j.MDC;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MdcLogEnhancerFilter implements Filter {
+
+  @Override
+  public void doFilter(
+      ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain)
+      throws IOException, ServletException {
+    HttpServletRequest httpRequest = (HttpServletRequest) servletRequest;
+
+    String requestId =
+        ((requestId = httpRequest.getHeader("Request-Id")) != null)
```

Comment:
`getHeader` is case-insensitive, fortunately

Reply:
could this be `X-Request-ID` or `X-Correlation-ID`, maybe even look for both before generating one ourselves? https://en.wikipedia.org/wiki/List_of_HTTP_header_fields